src/map.js

```diff
@@ -15,21 +16,23 @@ export default function map (node, callback, o) {
 }
 
 function _map (node, callback, o = {}, property, parent) {
-	let ignored = o.except && matches(node, o.except);
+	if (Array.isArray(node)) {
+		return node.map(n => _map(n, callback, o, property, parent));
+	}
 
-	if (!ignored && matches(node, o.only)) {
-		let ret = callback(node, property, parent);
+	const ignore = o.except && matches(node, o.except);
+	const explore = !ignore && matches(node, o.only);
+	let copiedNode = {...node};
```

Comment:
Why did you move this here? I had it where I had it so that we don't clone unless it's actually needed.
Btw if you're ok with my changes, instead of porting them here and I try to figure out what you changed and why, another option would be for you to push the tests to my branch (or merge and reduce this PR to the tests and the other fix).